week 1/lists_n_loops/index_search.py

```diff
@@ -0,0 +1,79 @@
+"""
+Group of functions that search all instances of a key in a list.
+Functions do same thing but with different looping mechanism.
+"""
+
+
+def search_key_first(nums: list[int], key: int) -> list[int]:
+    """
+    Traversal with a ranged for loop
+    :param nums: list of integers
+    :param key: integer to search for
+    :return: all indexes of the key
+    """
+    indexes = []
+
+    for i in range(len(nums)):
+        if nums[i] is key:
+            indexes.append(i)
+    return indexes
+
+
+def search_key_second(nums: list[int], key: int) -> list[int]:
+    """
+    Traversal with a standard list while loop
+    :param nums: list of integers
+    :param key: integer to search for
+    :return: all indexes of the key
+    """
+    indexes = []
+    length = len(nums)
+    i = 0
+
+    while i < length:
+        if nums[i] is key:
```

Comment:
`is` is identity operator, not equality. For integers, this might not have an impact but in terms of objects and other data types, you can expect some unexpected behavior with `is`.